keras/backend/tensorflow_backend.py

```diff
@@ -3400,7 +3401,8 @@ def _preprocess_conv1d_input(x, data_format):
     # Returns
         A tensor.
     """
-    if dtype(x) == 'float64':
+    if (dtype(x) == 'float64'
```

Comment:
Maybe for readability, we could have a simple function called `_needs_casting(x)`. Then those if statements would become `if _needs_casting(x):`. would it be then more readable?

Reply:
I think it's hard to describe the situation here and `_needs_casting` is not specific enough. Therefore I would prefer using comment.

Reply:
Sure then. It was only a suggestion.